staging/cse/windows/kubernetesfunc.ps1

```diff
@@ -129,9 +129,68 @@ function Write-KubeClusterConfig {
         Destination = "c:\k";
     }
 
+    Update-KubeletConfigArgsForDualStack -ClusterConfiguration $Global:ClusterConfiguration
+
     $Global:ClusterConfiguration | ConvertTo-Json -Depth 10 | Out-File -FilePath $global:KubeClusterConfigPath
 }
 
+# Update-KubeletConfigArgsForDualStack updates the $Global:ClusterConfiguration object
+# to append the --node-ip argument for kubelet if the cluster is dualstack and the
+# host is configured with an IPv4 and IPv6 on the interface using the default route.
+# If the host does not have both IPs it will raise erro WINDOWS_CSE_ERROR_DUAL_STACK_NO_EXACT_TWO_IPS
+function Update-KubeletConfigArgsForDualStack {
+    param(
+        [Parameter(Mandatory = $true)][PSCustomObject]
+        $ClusterConfiguration
+    )
+
+    # validate ClusterConfiguration is in expected format
+    if (($ClusterConfiguration.Kubernetes -eq $null) -or `
+        ($ClusterConfiguration.Kubernetes.Kubelet -eq $null)) {
+        return
+    }
+
+    # nothing to do if not a dualstack clsuter or the --node-ip is already in the arg list
+    if ((!$global:IsDualStackEnabled) -or `
+        (Test-NodeIPArgExists -Args $ClusterConfiguration.Kubernetes.Kubelet.ConfigArgs)) {
+        return
+    }
+
+    $defaultRouteIfIndex = (Get-NetRoute -DestinationPrefix 0.0.0.0/0).ifIndex

```

Comment:
it may be better to exit with an error code when `Get-NetRoute -DestinationPrefix 0.0.0.0/0` returns empty and `.ifIndex` will fail if there is some bug in vmss instance's network